aiohttp/client.py

```diff
@@ -329,6 +333,20 @@ def request(
         self, method: str, url: StrOrURL, **kwargs: Any
     ) -> "_RequestContextManager":
         """Perform HTTP request."""
+
+        valid_methods = [
+            "GET",
+            "POST",
+            "PUT",
+            "DELETE",
+            "HEAD",
+            "OPTIONS",
+            "TRACE",
+            "CONNECT",
+        ]
+        if method.upper() not in valid_methods:
+            raise ValueError(f"Invalid HTTP method: {method}")
```

Comment:
There are dozens of know HTTP methods. The spec does not dissallow others, only that they are the correct format. We definitely check this format in the parser for incoming requests with a regex, but I can't remember if there's any validation on outgoing requests.